cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,232 @@
+
+8. Deb deletes the resource provider Mycompany.Messaging
+
+    ```bash
+    rad resource-provider delete Mycompany.Messaging
+    Warning: `myapp` application has resources of type `Mycompany.Messaging/plaidResource` provisioned and running. Are you sure you want to delete the resource provider Mycompany.Messaging? (yes/no)
+    ```
+
+## Finalized set of commands
+
+### rad resource-provider
+
+#### 1. rad resource-provider create
+
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Creating a resource provider involves creating new resource types that can be used in your applications. Inputs can be passed in as a file or as flags.
+
+Create a resource provider from a file 
+    
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
+    Resource provider Mycompany.Messaging created successfully
+    ```
+**Flags**
+- `-f`, `--fom-file` : The file that contains the resource provider definition
+
+#### 2. rad resource-provider list
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Users can list all the resource providers available in your tenant.
+
+    ```bash
+    rad resource-provider list 
+    RESOURCE-PROVIDER   TYPE                                STATE
+    Mycompany.Messaging System.Resources/resourceProviders  Succeeded
+    ```
+
+#### 3. rad resource-provider show
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. User can display the details of the resource providers.
+
+    ```bash
+    rad resource-provider show Mycompany.Messaging
+    RESOURCE-PROVIDER   TYPE                                STATE
+    Mycompany.Messaging System.Resources/resourceProviders  Succeeded
+    ```
+
+#### 4. rad resource-provider delete
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Users can delete a resource provider.
+
+    ```bash
+    rad resource-provider delete Mycompany.Messaging
+    ```
+**Behaviors**
+Option 1 : Radius can prompt the user with a warning message if there are resources provisioned and running under the resource provider. If the user confirms deletion, Radius cleans up everything, deletes resources, types and resource providers.
+
+Option 2 : Radius can error out if there are resources provisioned and running under the resource provider. The user needs to delete the resources before deleting the resource provider.
+
+### rad resource-type
+
+#### 1. rad resource-type create
+Resource types are the entities that implement resource types such as 'Applications.Core/containers'. Each resource type can define multiple API versions, and each API version defines a schema that resource instances conform to. Inputs can be passed in as a file or as flags.
+
+Create a resource type from a file 
+    
+```bash
+rad resource-type create Mycompany.Messaging/plaidResource -f plaid.yaml
+```
```

Comment:
Do we need both `rad resource-provider create` and `rad resource-type create`? what would happen if we just built one of these?

Also, is this the same file as was used in `rad resource-provider create`?

As much as possible, let's create shared investments.

Reply:
yes same behavior across `rad resource-provider create` and `rad resource-type create`. We expect `rad resource-type create` to be the command a user interacts with and will be featured in the docs